### PR TITLE
Nginx pagespeed

### PR DESCRIPTION
**Pull request by vaibhavnoob**
This guide explains how to install Nginx with ngx_pagespeed module which is created by Google.

### CODE REVIEW

**Comment by vaibhavnoob** [comment]
any update?

**Comment by thetechstewart** [comment]
@vaibhavnoob: Your guide will be up for a tech test by a member of the Linode Writing team before the end of this week.  Once we have completed the tech test on your guide, we will follow up with you regarding any additional steps.

File: docs/websites/nginx/nginx-and-pagespeed-on-ubuntu-14-04.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
+---
+author:
+    name: Linode Community
```
Comment (felicianotech) [inline docs/websites/nginx/nginx-and-pagespeed-on-ubuntu-14-04.md]: The YAML code at the top of the guide is incorrect. A quick look tells me that the issue may be spacing. YAML specifically uses 2 spaces to indent, tabs not allowed. I see this guide is using 4 spaces. That is causing the guide to break in our publishing system.

File: docs/websites/nginx/nginx-and-pagespeed-on-ubuntu-14-04.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+ - '[Google PageSpeed Filter Docs](https://developers.google.com/speed/pagespeed/module/config_filters)'
+---
+
+This document describes how you can install Nginx (pronounced engine-x) with the ngx_pagespeed module. PageSpeed module helps us to minify CSS, javascript, images, remove whitespaces from HTML and many other features.
+
+Make sure that before starting this guide you have read through and completed our [Getting Started](/docs/getting-started/) guide.
+
+##Set the hostname
+
+Before you install any packages, ensure that your hostname is correct by completing the [Setting Your Hostname](/docs/getting-started#sph_setting-the-hostname) section of the Getting Started guide. Issue the following commands to verify:
```
Comment (felicianotech) [inline docs/websites/nginx/nginx-and-pagespeed-on-ubuntu-14-04.md]: The anchor link to the Setting The Hostname part of the Getting Started Guide is incorrect. You can use `https://linode.com/docs/getting-started#setting-the-hostname` for the entire section or maybe even link directly to the section relavent to Ubuntu 14.04 `https://linode.com/docs/getting-started#debian-7--slackware--ubuntu-1404`.

File: docs/websites/nginx/nginx-and-pagespeed-on-ubuntu-14-04.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    hostname
+    hostname -f
+
+##System Setup
+
+Make sure your system is up to date using apt:
+
+    sudo apt-get update
```
Comment (felicianotech) [inline docs/websites/nginx/nginx-and-pagespeed-on-ubuntu-14-04.md]: We're missing the upgrade portion here. Also, Ubuntu (and Debian 8) now have the plain apt command for added legibility. The two commands could be combined.

`apt update && apt upgrade`

or

`apt-get update && apt-get upgrade`

File: docs/websites/nginx/nginx-and-pagespeed-on-ubuntu-14-04.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+This ensures that all software is up to date and running the latest version.
+
+##Installing required packages
+
+Pagespeed module requires some extra packages which you should install to make it run properly, issue the following command in the terminal.
+
+    sudo apt-get install build-essential zlib1g-dev libpcre3 libpcre3-dev unzip
```
Comment (felicianotech) [inline docs/websites/nginx/nginx-and-pagespeed-on-ubuntu-14-04.md]: For many of the commands, I see you've prefixed the commands with `sudo`. The guide currently has the user running as the root user. You can stick with the root user and drop the `sudo` prefix from all of these commands or you can walk them through setting up a normal user, logging in with that user, and then leave `sudo` on these commands.

File: docs/websites/nginx/nginx-and-pagespeed-on-ubuntu-14-04.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+##Download ngx_pagespeed module
+
+After installing necessary packages, you need to download ngx_pagespeed module source. Issue the following command in the terminal one by one.
+
+    cd
+    NPS_VERSION=1.9.32.6
+    wget https://github.com/pagespeed/ngx_pagespeed/archive/release-${NPS_VERSION}-beta.zip
```
Comment (felicianotech) [inline docs/websites/nginx/nginx-and-pagespeed-on-ubuntu-14-04.md]: The URL in this command returns HTTP 404. Will you please investigate?

**Comment by felicianotech** [comment]
Hi,

My name is Ricardo and I have initiated the tech edit for this guide. I have some things that need to be addressed from my first round of tech testing. Most of my notes were made as inline comments that you can see above or from the 'Files changed' tab.

Please let us know when your pull request (PR) has been updated to address my notes. Also, if you have any questions please let us know. Thank you.

-Ricardo

**Comment by vaibhavnoob** [comment]
Updated the guide :+1:, I am not getting any 404 response 
screenshot from 2015-09-26 11 12 33

**Comment by felicianotech** [comment]
Will do another run through, thanks.

**Comment by vaibhavnoob** [comment]
OK :)

**Comment by cwlinode** [comment]
@vaibhavnoob: Thanks for getting this back to us. I'm continuing with the tech test but having an issue with the Pagespeed options. After adding the lines given into nginx.conf, attempting to start nginx gives the error:
`nginx: [emerg] unknown directive "2}\.[^.]" in /usr/local/nginx/conf/nginx.conf:82`.

Is this something you encountered at any point in your own testing?

Here's the relevant part of my server.conf:
```
    server {
        listen       80;
        . . . 
        #location ~ /\.ht {
        #    deny  all;
        #}

        pagespeed on;
        pagespeed FileCachePath /var/ngx_pagespeed_cache;
        location ~ “\.pagespeed\.([a-z]\.)?[a-z]{2}\.[^.]{10}\.[^.]+” {
        add_header “” “”;
        }
        location ~ “^/pagespeed_static/” { }
        location ~ “^/ngx_pagespeed_beacon$” { }

    }
```

Additionally, we like to encourage our readers to do administrative tasks as a non-root user. Could you please add `sudo` to the relevant commands in your guide.

Thank you, and I look forward to testing your next draft.

**Comment by alexfornuto** [comment]
I've only glanced at this guide, but I notice immediately that sections with multiple steps should have the steps numbered for better clarity, per our style guide: https://linode.com/docs/style-guide#format

**Comment by vaibhavnoob** [comment]
@cwlinode: I am not getting any error. @alexfornuto: Could you please check it again?

**Comment by alexfornuto** [comment]
@vaibhavnoob I believe that both @felicianotech and @cwlinode's issues were cause by some non-unix-standard formatting, invisible to the human eye but problematic for the guides users, as well as the guide itself when tested on our platform. 

I have a branch ( https://github.com/alexfornuto/docs/tree/nginx-pagespeed ) which solves this issues and has other formatting fixes, but for an unknown reason I am unable to create a PR on your fork. Please PR and accept these changes into your fork, at which point we can discuss some other changes to make to the guide.

**Comment by vaibhavnoob** [comment]
Hey I am new to Github, what I need to do? Can you please create a PR again on my fork?

**Comment by alexfornuto** [comment]
No I can't. As far as I can tell you have some restrictions on your account or your fork, but you're not in the list of forks I can create a PR to. You can follow the link above to my branch, and create your own PR from my fork to yours.

**Comment by vaibhavnoob** [comment]
Done. Please check again.

**Comment by alexfornuto** [comment]
Looking good! My next concern with this guide is that it doesn't show any practical use of Pagespeed, or even any way to test its functionality. Adding the module and compiling are obviously critical steps, but the guide is incomplete unless it shows you how to use what you've just installed.

**Comment by vaibhavnoob** [comment]
Thanks for the reply. After following these steps PageSpeed module will automatically enable CoreFilters which contain the following filters:
   add_head
   combine_css
   combine_javascript
   convert_meta_tags
   extend_cache
   fallback_rewrite_css_urls
   flatten_css_imports
   inline_css
   inline_import_to_link
   inline_javascript
   rewrite_css
   rewrite_images
   rewrite_javascript
   rewrite_style_attributes_with_url

**Comment by alexfornuto** [comment]
@vaibhavnoob Do you feel that the information in the comment above belongs in the guide?

**Comment by alexfornuto** [comment]
Asking @kcraig34 to provide a copy edit.

**Comment by kcraig34** [comment]
@vaibhavnoob - copy edit completed, awaiting your merge. See: https://github.com/vaibhavnoob/docs/pull/2/files

**Comment by vaibhavnoob** [comment]
@kcraig34: Done

**Comment by cwlinode** [comment]
@vaibhavnoob  I have a final draft with some light metadata updates but am unable to create a PR against this branch. Please correct so we can get this published.

**Comment by vaibhavnoob** [comment]
@cwlinode This is your branch (https://github.com/cwlinode/docs/tree/pagespeed)? I will create PR myself.

**Comment by vaibhavnoob** [comment]
@cwlinode done

**Comment by cwlinode** [comment]
@vaibhavnoob  You're published:
https://www.linode.com/docs/websites/nginx/nginx-with-pagespeed-on-ubuntu-14-04

Though I can't seem to find a Paypal address for you in past emails. Please send that info to  contribute@linode.com and we'll get the payment out asap.

**Pull request merged**